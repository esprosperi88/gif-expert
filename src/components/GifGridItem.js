
import PropsTypes from 'prop-types';

export const GifGridItem =( { url, title}) =>{

    return  <div className="card animate__animated animate__fadeIn">
                <p>{title}</p>
                <img src={url} alt={title}/>
            </div>;           


} 

GifGridItem.propTypes={
    url: PropsTypes.string.isRequired,
    title: PropsTypes.string.isRequired,
}


export default GifGridItem;