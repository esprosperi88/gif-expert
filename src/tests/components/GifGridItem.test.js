import React from "react";
import GifGridItem from '../../components/GifGridItem';
import {shallow}   from 'enzyme'


describe('Prueba component - GifGridItem',() => {
    const key ="categoria";
    const url="(url)";
    const title="(titulo)";

    let wrapper;

    beforeEach( () => {        
        wrapper = shallow(<GifGridItem 
            key={key}
            url={url}
            title={title} />);
    });


    test('GifGridItem - snapshot',() => {
        expect (wrapper).toMatchSnapshot();
    });

    test('GifGridItem - verifica titulo ',() => {
        const titulo = wrapper.find('p').text();


        expect (titulo.trim()).toBe(title);


    });
    test('GifGridItem - verifica url y alt',() => {
        const img = wrapper.find('img');

        


        expect (img.prop('src')).toBe(url);
        expect (img.prop('alt')).toBe(title);


    });
    test('GifGridItem - verifica clase CSS  ',() => {
        const div = wrapper.find('div');

        console.log(div);


        expect (div.prop('className').includes('animate__fadeIn')).toBe(true);


    });


});