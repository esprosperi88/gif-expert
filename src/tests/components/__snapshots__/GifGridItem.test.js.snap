// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba component - GifGridItem GifGridItem - snapshot 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    (titulo)
  </p>
  <img
    alt="(titulo)"
    src="(url)"
  />
</div>
`;
